eZ/Bundle/EzPublishCoreBundle/URLChecker/Handler/HTTPHandler.php

```diff
@@ -0,0 +1,130 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace eZ\Bundle\EzPublishCoreBundle\URLChecker\Handler;
+
+use eZ\Publish\API\Repository\Values\URL\URL;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+class HTTPHandler extends AbstractURLHandler
+{
+    /**
+     * {@inheritdoc}
+     *
+     * Based on https://www.onlineaspect.com/2009/01/26/how-to-use-curl_multi-without-blocking/
+     */
+    public function validate(array $urls)
+    {
+        if (!$this->options['enabled']) {
+            return;
+        }
+
+        $master = curl_multi_init();
```

Comment:
Can't [GuzzleHttp](https://github.com/guzzle/guzzle) handle that (meaning to get rid of all direct calls to `curl_`)?

Reply:
Because is enough for our needs. curl_multi_* gives as ability to check URLs asynchronous, which is crucial to do this efficiently. Can be improved in future 😉

Reply:
Note that we don't require curl in kernel (we do for full install)

Reply:
> Note that we don't require curl in kernel (we do for full install)

That's true. The only solution for now which I can see is detecting if curl is available at runtime and throw exception/log error "Curl is required to check URLs with http://(s) schema" if doesn't.   What should we do?

Reply:
> Note that we don't require curl in kernel (we do for full install)

True, I think behat uses it.

So this is fine then :+1:

Reply:
I was implying adding ext-curl to composer.json if possibler

Reply:
@andrerom Done 5ea716a65dd247a513a86a0a6dbfa81f57f70ee7